src/master-playlist-controller.js

```diff
@@ -352,8 +352,8 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     // if we don't have any demuxed audio tracks
     // return an empty array
-    if (!master && !master.mediaGroups && !master.mediaGroups.AUDIO) {
-      return [];
+    if (!master || !master.mediaGroups || !master.mediaGroups.AUDIO) {
+      return master && master.playlists;
```

Comment:
If there's no master, is it OK that this returns `undefined`? Would it be better to return `[]`?

The comment above may need to be changed too otherwise. And would returning `master.playlists` end up returning non audio only tracks?

Reply:
I will make it return an empty array without a master. It will could end up returning non-audio playlists but if we don't have audio groups and this function is called, which should only happen if we detect that the playlist is audio only, Then we should just return the main playlists.